miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/RunScannerClient.java

```diff
@@ -0,0 +1,113 @@
+package uk.ac.bbsrc.tgac.miso.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Semaphore;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+
+import uk.ac.bbsrc.tgac.miso.core.util.LatencyHistogram;
+import uk.ac.bbsrc.tgac.miso.dto.NotificationDto;
+import uk.ac.bbsrc.tgac.miso.dto.ProgressiveRequestDto;
+import uk.ac.bbsrc.tgac.miso.dto.ProgressiveResponseDto;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+
+@Service
+@Transactional(rollbackFor = Exception.class)
+public class RunScannerClient {
```

Comment:
I can move this service, but since `LatencyHistogram` is also used by the run scanner itself, it can't be moved.